### PR TITLE
docs: doc update for mempoolfullrbf default + log deprecation

### PR DESCRIPTION
**Pull request by glozow**
Followup to #30493. Update bips.md and policy/*.md to reflect new default rules around signaling requirements in RBF.
Also, log a warning when `-mempoolfullrbf=0` that this config option is deprecated and will be removed in a future release.

### CODE REVIEW

**Comment by petertodd** [comment]
ACK 7168b45ba2fa41737bdecbf102e8d61e205fcd9c

File: doc/policy/mempool-replacements.md [hunk 1]
```diff
@@ -10,14 +10,12 @@ A transaction ("replacement transaction") may replace its directly conflicting t
 their in-mempool descendants (together, "original transactions") if, in addition to passing all
 other consensus and policy rules, each of the following conditions are met:
 
-1. The directly conflicting transactions all signal replaceability explicitly. A transaction is
+1. If `-mempoolfullrbf=0` (not true by default), the directly conflicting transactions all signal replaceability explicitly. A transaction is
```
Comment (instagibbs) [inline doc/policy/mempool-replacements.md:13]:
```Suggestion
1. If `-mempoolfullrbf=0` (the value is 1 by default), the directly conflicting transactions all signal replaceability explicitly. A transaction is
Comment

```

File: doc/policy/packages.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ The following rules are enforced for all packages:
 
 * Only limited package replacements are currently considered. (#28984)
 
-   - All direct conflicts must signal replacement (or the node must have `-mempoolfullrbf=1` set).
+   - If `-mempoolfullrbf=0` (not true by default), all direct conflicts must signal replacement.
```
Comment (instagibbs) [inline doc/policy/packages.md:41]:
```Suggestion
    - If `-mempoolfullrbf=0` (the value is 1 by default), all direct conflicts must signal replacement.
```

File: src/node/mempool_args.cpp [hunk 3]
```diff
@@ -93,6 +93,9 @@ util::Result<void> ApplyArgsManOptions(const ArgsManager& argsman, const CChainP
     }
 
     mempool_opts.full_rbf = argsman.GetBoolArg("-mempoolfullrbf", mempool_opts.full_rbf);
+    if (!mempool_opts.full_rbf) {
+        LogPrintf("Warning: mempoolfullrbf=0 set but deprecated and will be removed in a future release\n");
```
Comment (tdb3) [inline src/node/mempool_args.cpp:97]: Maybe we can use `LogInfo` here?

**Review by tdb3 — COMMENTED** [review]
Approach ACK

File: doc/policy/mempool-replacements.md (hunk 1 above)
Comment (glozow) [reply doc/policy/mempool-replacements.md]: done

File: doc/policy/packages.md (hunk 2 above)
Comment (glozow) [reply doc/policy/packages.md]: done

File: src/node/mempool_args.cpp (hunk 3 above)
Comment (glozow) [reply src/node/mempool_args.cpp]: right of course! done, thanks

**Review by tdb3 — APPROVED** [review]
ACK 1f93e3c360f3da38a02fe6c01551f424a2d63dc9

**Comment by instagibbs** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/30594/commits/1f93e3c360f3da38a02fe6c01551f424a2d63dc9

**Pull request merged**